File: app/src/main/java/com/aravindh/andriodbasesetup/di/model/Car.java
package com.aravindh.andriodbasesetup.di.model;

import android.util.Log;

import javax.inject.Inject;

import static com.aravindh.andriodbasesetup.di.DaggerActivityKt.DAGGER;

/**
 * Created by aravindh.s on 23-02-2020.
 */
public class Car {


    /*FIELD INJECTION TRIGGER 2nd*/
    @Inject
    Engine engine;

    @Inject
    Wheels wheels;

    /*CONSTRUCTOR INJECTION TRIGGER 1st*/
    @Inject
    public Car() {

    }

    public void drive() {
        engine.start();
        Log.d(DAGGER, "driving...");
    }

    /*METHOD INJECTION TRIGGER 3rd*/
    @Inject
    public void getEngineNumber() {
        Log.d(DAGGER, "engineNumber is : 157563836853");
    }

}

File: app/src/main/java/com/aravindh/andriodbasesetup/dagger/Car.kt
package com.aravindh.andriodbasesetup.dagger

import javax.inject.Inject

class Car @Inject constructor(private val engine: Engine, private val wheels: Wheels) {

    fun drive(msg: String): String {
        return "$msg driving...!"
    }

    @Inject
    fun enableRemote(remote: Remote) {
        remote.setListener(this)
    }
}
